pywr-core/src/timestep.rs

```diff
@@ -1,23 +1,61 @@
-use chrono::{Duration, Months, NaiveDateTime};
+use chrono::{Months, NaiveDateTime, TimeDelta};
 use polars::datatypes::TimeUnit;
 use polars::time::ClosedWindow;
 use pyo3::prelude::*;
 use std::ops::Add;
 
 use crate::PywrError;
 
+#[pyclass]
+#[derive(Debug, Copy, Clone)]
+pub struct PywrDuration(TimeDelta);
+
+impl From<TimeDelta> for PywrDuration {
+    fn from(duration: TimeDelta) -> Self {
+        Self(duration)
+    }
+}
+
+impl PartialEq<TimeDelta> for PywrDuration {
+    fn eq(&self, other: &TimeDelta) -> bool {
+        self.0 == *other
+    }
+}
+
+impl PywrDuration {
+    pub fn days(days: i64) -> Self {
+        Self(TimeDelta::days(days))
+    }
+
+    pub fn whole_days(&self) -> Option<i64> {
+        if self.fractional_days() % 1.0 == 0.0 {
```

Comment:
Does this pass clippy checks of comparing against floating point values?

Reply:
I don't get a warning locally. Is there going to be an issue with floating point precision here? Might be better to use the `float_cmp` crate?

Reply:
I was possible expecting clippy to show this: https://rust-lang.github.io/rust-clippy/master/index.html#/float_cmp_const

However, it is part of the "restriction" group which means it should be considered case-by-case. The question is whether there's any reasonable output of `fractional_days()` which we expect to be treated as a whole day, but would not be by this test? Leap seconds? Are those still a thing?

If there's not already just add a test for this method.

Reply:
@Batch21 I did a bit of reading here. I think modulo on floats is a bit weird sometimes. It might be better to do the modulo on the `num_seconds()` as a `i64`. 

Something like this might be better?

```rust
const SECS_IN_DAY: i64 = 60 * 60 * 24;

  pub fn whole_days(&self) -> Option<i64> {
      if self.0.num_seconds() % SECS_IN_DAY == 0 {
           Some(self.0.num_days())
      } else {
           None
      }
  }

  pub fn fractional_days(&self) -> f64 {
      self.0.num_seconds() as f64 / SECS_IN_DAY as f64
  }
```